.github/workflows/build.yml

```diff
@@ -46,30 +46,48 @@ jobs:
       env:
         BASE: origin/${{ github['base_ref'] }}
       run: |
+        model_paths=(
+          fhir-model
+        )
+        validation_paths=(
+          fhir-validation
+          fhir-path/src/main
+        )
+        search_paths=(
+          fhir-search
+        )
         jdbc_paths=(
+          fhir-persistence-jdbc
           fhir-search/src/main
           fhir-persistence/src/main
           fhir-persistence-schema/src/main
-          fhir-persistence-jdbc
-        )
-        model_paths=(
-          fhir-model
         )
         declare -a PROFILES_ARR
 
-        if [ -n "$(git --no-pager diff --name-only ${BASE} ${GITHUB_SHA} -- ${jdbc_paths[@]})" ]; then
-          echo "Changes affect fhir-persistence-jdbc, running ALL"
-          PROFILES_ARR+=(jdbc-all-tests)
-        else
-          echo "No changes affect fhir-persistence-jdbc, running MINIMAL"
-        fi
-
         if [ -n "$(git --no-pager diff --name-only ${BASE} ${GITHUB_SHA} -- ${model_paths[@]})" ]; then
           echo "Changes affect fhir-model, running ALL"
           PROFILES_ARR+=(model-all-tests)
         else
           echo "No changes affect fhir-model, running MINIMAL"
         fi
+        if [ -n "$(git --no-pager diff --name-only ${BASE} ${GITHUB_SHA} -- ${validation_paths[@]})" ]; then
+          echo "Changes affect fhir-validation, running ALL"
+          PROFILES_ARR+=(validation-all-tests)
+        else
+          echo "No changes affect fhir-validation, running MINIMAL"
+        fi
+        if [ -n "$(git --no-pager diff --name-only ${BASE} ${GITHUB_SHA} -- ${search_paths[@]})" ]; then
+          echo "Changes affect fhir-search, running ALL"
+          PROFILES_ARR+=(search-all-tests)
+        else
+          echo "No changes affect fhir-search, running MINIMAL"
+        fi
+        if [ -n "$(git --no-pager diff --name-only ${BASE} ${GITHUB_SHA} -- ${jdbc_paths[@]})" ]; then
+          echo "Changes affect fhir-persistence-jdbc, running ALL"
+          PROFILES_ARR+=(jdbc-all-tests)
+        else
+          echo "No changes affect fhir-persistence-jdbc, running MINIMAL"
+        fi
 
         if [ ${#PROFILES[@]} -eq 0 ]; then
           # Convert the PROFILES_ARR array to a comma-delimited string
```

Comment:
I think we'd better also enable the ability of cross check, e.g, (1) add fhir-example to the loop, and whenever fhir-example changes, maybe we should run all.txt for validation, jdbc etc.  (2) whenever model-path change happens, maybe we should run all.txt for jdbc or validation(as we discussed, there is no validator in fhir-model project at present).

Reply:
on (1) I totally agree, I'll add fhir-example to all 4 paths
on (2), this PR adds the logic so that if anything under `fhir-validation` or `fhir-path/src/main` changes, then we'll run the validation test with the ALL_JSON index

Reply:
Should we add fhir-model project to validation-paths? I mean e.g, if resource classes in it is updated or re-generated, should we trigger validation for all examples?

Reply:
sure, i can add that

Reply:
thanks, just try to make the matrix better.